gradle/shadow.gradle

```diff
@@ -0,0 +1,90 @@
+import org.gradle.api.internal.file.FileResolver
+
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import com.github.jengelman.gradle.plugins.shadow.tasks.DefaultInheritManifest
+import org.gradle.api.internal.file.FileResolver
+
+buildscript {
+    repositories { jcenter() }
+    dependencies {
+        classpath 'com.github.jengelman.gradle.plugins:shadow:1.2.4'
+    }
+}
+
+configure([project(':bson'), project(':mongo-java-driver')]) {
+    apply plugin: 'osgi'
+    apply plugin: 'com.github.johnrengelman.shadow'
+
+    configurations {
+        compile.extendsFrom shadow
+    }
+
+    shadowJar {
+        from sourceSets.main.output
+        configurations = [project.configurations.compile]
+        mergeServiceFiles()
+        relocate 'com.fasterxml', 'org.bson.shaded.fasterxml'
```

Comment:
WIth the latest commit I'm now seeing a META-INF directory containing just the MANIFEST.MF, which no longer has a Class-Path entry.  So all is well.